### PR TITLE
[CLD-7845] First pass at a README with some instructions for running the project

### PR DESCRIPTION
**Pull request by nickmisasi**
#### Summary
Adds some pizzazz to the homepage of the project with a README file as well as a DEVELOPERS file with some instructions for getting the project going. I've left a few TODO's for things like screenshots, and link outs to places like releases or docs.mattermost.com as those mature. 

#### Ticket Link
https://mattermost.atlassian.net/browse/CLD-7845

### CODE REVIEW

**Comment by nickmisasi** [comment]
@gabrieljackson want to run this by you for what we should cover content-wise and then loop Carrie in for a proper proofread after any changes

File: DEVELOPERS.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+```
+
+This allows you to specify a custom path for the state file, ensuring that your setup can accommodate different environments or configurations as needed.
+
+**NOTE**: MCNB must have write access to the directory where the state file exists in order to be able to persist your configuration between server restarts.
+
+## General Guidelines
+
+- Please ensure your code follows the project's coding conventions and style guidelines.
+- Write clear and concise commit messages that describe your changes.
```
Comment (gabrieljackson) [inline DEVELOPERS.md:77]:
```suggestion
- Write clear and concise commit messages that describes your changes.
```

**Review by gabrieljackson — APPROVED** [review]
The content looks good to me for a first pass!

File: DEVELOPERS.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
+
+# Contributing to Mattermost CloudNative Bootstrapper
```
Comment (cwarnermm) [inline DEVELOPERS.md:2]: I see a mix of gerunds (verbs that end in `ing`) and regular verbs across these section titles. No `ing` is preferred for verbs for clarity (as gerund verbs isn't a universal thing across all languages).

**Review by cwarnermm — APPROVED** [review]
This is a fantastic set of docs, @nickmisasi! Thank you so much! Left editorial feedback re: verb constructions. Happy to review again once the to-dos are addressed.

File: DEVELOPERS.md (hunk 2 above)
Comment (nickmisasi) [reply DEVELOPERS.md]: I've learned a new word today! Removed the "gerunds" :)

**Pull request merged**